sprintf.c

```diff
@@ -519,8 +519,11 @@ rb_str_format(int argc, const VALUE *argv, VALUE fmt)
 	VALUE sym = Qnil;
 
 	for (t = p; t < end && *t != '%'; t++) ;
-	if (t + 1 == end) ++t;
 	PUSH(p, t - p);
```

Comment:
To exit the loop after a single `'%'` at the end.
Otherwise, `p` skips it and reaches `end`.
This isn't a problem right now, but if `SHARABLE_MIDDLE_SUBSTRING` in string.c is non-zero, `*end` may not be `'\0'`.